proto/p4/config/p4info.proto

```diff
@@ -192,7 +192,15 @@ message Table {
   // added in the future
   repeated uint32 direct_resource_ids = 7;
   int64 size = 8;  // max number of entries in table
-  bool with_entry_timeout = 9;  // entry ageing is enabled for table
+  // this enum can be extended in the future with other behaviors, such as
+  // "HARD_EVICTION"
+  enum IdleTimeoutBehavior {
+    UNSPECIFIED = 0;
+    NO_TIMEOUT = 1;
+    NOTIFY_CONTROL = 2;
+  }
+  // is idle timeout supported for this table?
+  IdleTimeoutBehavior idle_timeout_behavior = 9;
```

Comment:
Is a target free to fail any P4 program defining a table with NOTIFY_CONTROL at compile time if, for example, the hardware provides insufficient timeout support?

Reply:
I'll make that clear in the spec.  Maybe we need a general statement there about when a compiler backend should reject a program vs when to use runtime errors.